pallets/services-payment/src/lib.rs

```diff
@@ -342,16 +290,54 @@ impl<T: Config> AuthorNotingHook<T::AccountId> for Pallet<T> {
         _block_number: BlockNumber,
         para_id: ParaId,
     ) -> Weight {
-        let total_weight = T::DbWeight::get().reads_writes(1, 1);
+        if Pallet::<T>::burn_free_credit_for_para(&para_id).is_err() {
+            let (amount_to_charge, _weight) = T::ProvideBlockProductionCost::block_cost(&para_id);
+            match T::Currency::withdraw(
+                &Self::parachain_tank(para_id),
+                amount_to_charge,
+                WithdrawReasons::FEE,
+                ExistenceRequirement::AllowDeath,
```

Comment:
Shouldn't this be KeepAlive? In the runtime we check that this account has enough credit with keep alive true:

```
// The true refers to keepAlive
            Balances::can_withdraw(&pallet_services_payment::Pallet::<Runtime>::parachain_tank(*para_id), remaining_to_pay).into_result(true).is_ok()
```

Reply:
yeah it is pretty much irrelevant tbh as we should not enter this function if the account does not have enough money. In the runtime it is important that we know the account will pay for the credits and otherwise we dont assign collators, but once this funciton is reached, it is irrelevant if we do one or the other..

But I can put keepAlice to be consistent